### ISSUE TITLE
[GENERAL] Mob - NPC combat positionings.

### ISSUE THREAD

**Issue by Eirolla** [WoWManiaUK/Blackwing-Lair#2823]
NPC's must not attack us from max range, when we move in, they are moving to their max attack range. 

**What should happen:**

NPC's should stay at 1 position untill they are out of range. (like at wow-mania)

**Comment by Sammy974** [comment]
This is an odd issue, I was going to report it too but I couldn't find the right way to do it. It's something about the hitbox, I guess. When I'm tanking, I can push the boss/mob forward. He always face me and keeps a distance from me. This makes him move in a strange way, there's definitely something wrong.

**Comment by Rushor** [comment]
What is happening:

Basicly, every NPC try to face you when you are in combat when them.

What should happen:

NPC shouldn't care if they are in front of you, in your back or on your side.
When you have their agro, they just go at your melee range and hit regardless of your position.

Plus, they always are at maximum melee hit range. If you move forward them, they will just go backward.
If a NPC have 8m radius melee range. If you stand in that range whether 8m or 1m, they shouldn't move.

**Comment by Echinaceax** [comment]
looks like keader didnt want it @Rushor lol
ive reassigned back to you ...one of you can sort it :)

**Comment by Eirolla** [comment]
the problem is most likely somewhere in TargetedMovementGenerator.cpp

**Comment by Echinaceax** [comment]
@Rushor @Kitzunu - can we get this fixed ASAP 
even pets can do this exploit, needs fixing

**Comment by jburkiett** [comment]
bump

**Comment by Rushor** [comment]
yeah the pushing is actually a blizzlike feature, but it should not happen that often

**Comment by wow-mania** [comment]
As-is for now. I will try to reduce the aggro radius a bit.

**Comment by Rushor** [comment]
okay must be somewhere in chasemovementgen

**Comment by Rushor** [comment]
fresh from retail! https://gyazo.com/f48e1d269f6de1424e738bc6994a021a

the behaviour is blizzlike. they should always stay on maxange if you melee, but the time they reposition is like 3 sec delay always + the is some tolerance

**Comment by questbugreporter** [comment]
@Rushor the mob didn't move to stand in front of you, it only moved back when you moved inside it. The worst problem on reign isn't the pushing (although it's pretty bad), but the fact that the mobs always want to stand in front of you.
And they're definitely not supposed to stay at MAX range all the time, try it against something like Argaloth or literally any enemy with a giant hit box.

**Comment by Rushor** [comment]
yes i try to figure out where that part is handled. maybe @Ovahlord can help me? :o
